classes/cargo.bbclass

```diff
@@ -0,0 +1,112 @@
+inherit rust-common
+
+# Move CARGO_HOME from default of ~/.cargo
+export CARGO_HOME = "${WORKDIR}/cargo_home"
+
+# If something fails while building, this might give useful information
+export RUST_BACKTRACE = "1"
+
+DEPENDS += "\
+    rust-bin \
+    cargo-bin-native \
+"
+
+# Do build out-of-tree
+B = "${WORKDIR}/target"
+export CARGO_TARGET_DIR = "${B}"
+
+# When cross-compiling, explicitly set target. When natively compiling target
+# the BUILD system.
+RUST_TARGET = "${@rust_target(d, 'TARGET')}"
+RUST_TARGET_class-native = "${@rust_target(d, 'BUILD')}"
+
+# Additional flags passed directly to the "cargo build" invocation
+EXTRA_CARGO_FLAGS ??= ""
+
+# Space-separated list of features to enable
+CARGO_FEATURES ??= ""
+
+CARGO_DEBUG_DIR = "${B}/${RUST_TARGET}/debug"
+CARGO_RELEASE_DIR = "${B}/${RUST_TARGET}/release"
+WRAPPER_DIR = "${WORKDIR}/wrappers"
+
+CARGO_BUILD_FLAGS = "\
+    --verbose \
+    --manifest-path ${S}/Cargo.toml \
+    --target=${RUST_TARGET} \
+    --${@build_type(d)} \
+    ${@base_conditional('CARGO_FEATURES', '', '', '--features "${CARGO_FEATURES}"', d)} \
+    ${EXTRA_CARGO_FLAGS} \
```

Comment:
Maybe `CARGO_EXTRA_FLAGS` instead?  Or is this following an existing pattern?  If not, I prefer namespace prefixing.